org.lflang/src/org/lflang/generator/c/CGenerator.java

```diff
@@ -2571,6 +2571,9 @@ private void createMainReactorInstance() {
                     errorReporter.reportError("Main reactor has causality cycles. Skipping code generation.");
                     return;
                 }
+                if (hasDeadlines) {
+                    this.main.assignDeadlines();
+                }
```

Comment:
This is where deadlines are assigned and propagated. Later levels are also assigned. Still "assignLevels" are invoked a lot later, "just in case" it seems